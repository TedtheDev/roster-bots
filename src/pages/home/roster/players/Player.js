import React, { Component } from 'react';
import { connect } from 'react-redux';
import { updatePlayer } from '../../../../modules/players';
import PropTypes from 'prop-types';

import EditableInput from '../../../../components/inputs/EditableInput';

export class Player extends Component {
    constructor(props) {
        super(props);

        this.state = { playerFirstName: this.props.player.firstName, playerLastName: this.props.player.lastName };
    }

    onKeyPressSubmit = (event) => {
        if(event.key === 'Enter') {
            let updatedPlayer= Object.assign({}, this.props.player);
            updatedPlayer.firstName = this.state.playerFirstName;
            updatedPlayer.lastName = this.state.playerLastName;
            this.props.dispatch(updatePlayer(updatedPlayer));
        }
    }

    onChangePlayerFirstName = (event) => {
        this.setState({playerFirstName: event.target.value});
    }

    onChangePlayerLastName = (event) => {
        this.setState({playerLastName: event.target.value});
    }

    render() {
        return (
            <React.Fragment>
                <EditableInput 
                    error={this.props.duplicateError} 
                    onKeyPress={this.onKeyPressSubmit} 
                    type='text' 
                    onChange={this.onChangePlayerFirstName} 
                    value={this.state.playerFirstName} 
                />
                <EditableInput 
                    error={this.props.duplicateError} 
                    onKeyPress={this.onKeyPressSubmit} 
                    type='text' 
                    onChange={this.onChangePlayerLastName} 
                    value={this.state.playerLastName} 
                />

                <div>{this.props.player.strength + this.props.player.agility + this.props.player.speed} </div>
                <div> {this.props.player.strength}</div>
                <div> {this.props.player.agility}</div>
                <div> {this.props.player.speed}</div>
            </React.Fragment>
        )
    }
    
}

Player.propTypes = {
    duplicateError: PropTypes.bool.isRequired,
    player: PropTypes.shape({
        firstName: PropTypes.string.isRequired,
        lastName: PropTypes.string.isRequired,
        strength: PropTypes.number.isRequired,
        agility: PropTypes.number.isRequired,
        speed: PropTypes.number.isRequired
    })
};

export default connect(null)(Player);