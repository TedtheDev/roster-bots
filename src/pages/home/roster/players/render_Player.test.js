import React from 'react';
import renderer from 'react-test-renderer';
import { Player }  from './Player';

describe('<Player />', () => {
    it('renders correctly', () => {
        const player = {
            firstName: 'Frank',
            lastName: 'Frank',
            agility: 34,
            speed: 20,
            strength: 3,
            id: 'aaa222'
        }


        const wrapper = renderer.create(<Player player={player} duplicateError={false}/>);
        const tree = JSON.stringify(wrapper);

        expect(tree).toMatchSnapshot();
    });
});