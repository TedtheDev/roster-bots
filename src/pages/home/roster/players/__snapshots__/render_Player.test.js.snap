// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> renders correctly 1`] = `"[{\\"type\\":\\"input\\",\\"props\\":{\\"className\\":\\"sc-bdVaJa dUqDqt\\",\\"type\\":\\"text\\",\\"value\\":\\"Frank\\"},\\"children\\":null},{\\"type\\":\\"input\\",\\"props\\":{\\"className\\":\\"sc-bdVaJa dUqDqt\\",\\"type\\":\\"text\\",\\"value\\":\\"Frank\\"},\\"children\\":null},{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":[\\"57\\",\\" \\"]},{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":[\\" \\",\\"3\\"]},{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":[\\" \\",\\"34\\"]},{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":[\\" \\",\\"20\\"]}]"`;
